src/main/java/org/squiddev/plethora/integration/buildcraft/TileControllable.java

```diff
@@ -0,0 +1,53 @@
+package org.squiddev.plethora.integration.buildcraft;
+
+import buildcraft.api.tiles.IControllable;
+import buildcraft.core.BCCore;
+import dan200.computercraft.api.lua.LuaException;
+import org.squiddev.plethora.api.Injects;
+import org.squiddev.plethora.api.method.wrapper.FromTarget;
+import org.squiddev.plethora.api.method.wrapper.PlethoraMethod;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@Injects(BCCore.MODID)
+public final class TileControllable {
+
+	@PlethoraMethod(modId = BCCore.MODID, doc = "-- Get a list of all supported control modes.")
+	public static List<String> getSupportedControlModes(@FromTarget IControllable controllable) {
+		return Arrays.stream(IControllable.Mode.VALUES).filter(controllable::acceptsControlMode).map(mode -> mode.lowerCaseName).collect(Collectors.toList());
+	}
+
+	@PlethoraMethod(modId = BCCore.MODID, doc = "-- Get the control mode.")
+	public static String getControlMode(@FromTarget IControllable controllable) {
+		return controllable.getControlMode().lowerCaseName;
+	}
+
+	@PlethoraMethod(modId = BCCore.MODID, doc = "-- Reports true if the Block accept the given mode.")
+	public static boolean acceptsControlMode(@FromTarget IControllable controllable, String modeName) throws LuaException {
+		IControllable.Mode mode;
+		try {
```

Comment:
I _think_ you can use `IControllable.Mode` directly on the argument. We definitely do it for other enums:

https://github.com/SquidDev-CC/plethora/blob/527d945ab4c69514a63d2a04e2fcb767b8cfb3f8/src/main/java/org/squiddev/plethora/integration/rf/MethodsEnergy.java#L30